### PR TITLE
AzureIDIR (IDIR MFA) support

### PR DESCRIPTION
**Pull request by norrisng-bc**
# Description

This PR introduces full support for IDIR MFA (a.k.a. `azureidir`) as an identity provider, alongside the existing SiteMinder IDIR (or `idir`). 

`azureidir` was already implicitly supported, since `azureidir`-based JWTs also come from the Pathfinder SSO common Keycloak realm (this allows clients to supply their own JWTs). This PR "merges" the two identities, in the sense that whenever `azureidir` is present in the incoming JWT's `identity_provider` or in the query parameters, it is overwritten/stored as `idir` inside COMS.

Additionally, the behaviour in #311 has now been extended to also include IDP-based permissions. This allows such permissions to be inherited when a child bucket is created.

https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-3713
https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-4364

## Types of changes

Bug fix (non-breaking change which fixes an issue)
New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

After discussion, it was agreed that overwriting the `identity_provider` field at the auth middleware layer would be the simplest way to merge `idir` and `azureidir`. While this certainly isn't the cleanest, other methods would've entailed modifying significantly more code, edge case handling, updating tests etc.

### CODE REVIEW

File: app/src/components/constants.js [hunk 1]
```diff
@@ -29,7 +29,8 @@ module.exports = Object.freeze({
    * are given additional permissions.
   */
   ElevatedIdps: [
-    'idir'
+    'idir',
+    'azureidir'
```
Comment (TimCsaky) [inline app/src/components/constants.js]: dont think we need this if idp is always `idir` (?)

File: app/src/services/user.js
Comment (TimCsaky) [inline app/src/services/user.js:169]: I thought we were going to keep the value in db user.idp as `idir` (and not update to `azureidir`) when user logs in with azure idir.(?)
eg: around line 157: 
```
if (newUser.idp === 'azureidir') newUser.idp = 'idir';
```
?

Comment (TimCsaky) [reply app/src/services/user.js:169]: but thinking more..
client app (eg bcbox) would need to also need to re-elvaluate user token.idp as always `idir`

File: app/src/components/constants.js (hunk 1 above)
Comment (norrisng-bc) [reply app/src/components/constants.js]: I think I threw in there as a "just in case", but yeah it's probably a bit redundant and unnecessary.

File: app/src/services/user.js
Comment (norrisng-bc) [reply app/src/services/user.js:169]: The implicit `azureidir` to `idir` conversion is handled at the auth middleware layer, so the UserService won't even see it. The user will just appear to it as `idir` (even though the underlying JWT field is actually `azureidir`), so it's as if there's no change or update needed on `user.idp` in the COMS DB.

Comment (norrisng-bc) [reply app/src/services/user.js:169]: But yes, the client does need to make this same assumption and handle it accordingly (e.g. if the JWT `identity_provider` is `azureidir`, and you want to search for IDP-level permissions the user has access to, make the COMS API call with `idp=idir` instead - don't just use that field from the JWT as-is).

...that is, unless we update the COMS-side code in a couple more places so that `azureidir` is silently converted to `idir` (e.g. wherever `idp` is accepted as a parameter), in addition to `currentUser`.

**Review by TimCsaky — APPROVED** [review]
